packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1231,10 +1311,40 @@ class SemanticsOwner extends ChangeNotifier {
     super.dispose();
   }
 
+  // Updates the nextNodeId indices on the semantics nodes. These indices are used
+  // on the platform side to order the nodes for traversal by the accessibility
+  // services. If the nextNodeId index for a node changes, it will be marked as dirty.
+  void _updateTraversalOrder() {
+    final List<_TraversalSortNode> nodesInSemanticsTraversalOrder = <_TraversalSortNode>[];
+    SemanticsSortOrder currentSortOrder = new SemanticsSortOrder(keys: <SemanticsSortKey>[]);
+    bool visitor(SemanticsNode node) {
+      final SemanticsSortOrder previousOrder = currentSortOrder;
+      if (node.sortOrder != null) {
+        currentSortOrder = currentSortOrder.merge(node.sortOrder);
+      }
+      final _TraversalSortNode traversalNode = new _TraversalSortNode(node, currentSortOrder);
+      nodesInSemanticsTraversalOrder.add(traversalNode);
+      if (node.hasChildren) {
+        node.visitChildren(visitor);
+      }
+      currentSortOrder = previousOrder;
+      return true;
+    }
+    rootSemanticsNode.visitChildren(visitor);
+    nodesInSemanticsTraversalOrder.sort();
+    int nextNodeId = -1;
+    for (_TraversalSortNode node in nodesInSemanticsTraversalOrder.reversed) {
+      node.node._updateNextNodeId(nextNodeId);
```

Comment:
Can we document somewhere around the setter/getter for nextNodeId (or for _updateNextNodeId) what setting this to -1 means? Especially, does it mean: Not followed by any node? Or the order of the next node has not been defined?

Reply:
Done.  It means not followed by another node, and is the value that the last node in the list gets.